### ISSUE TITLE
Localized blocks field writes empty localized arrays to database, forcing fallback-locale to fail for that field

### ISSUE THREAD

**Issue by quornik** [payloadcms/payload#1695]
# Bug Report

This bug report is a fallowup to discussion item I started #1694

When the blocks field is localized, it writes itself to the database as object with localized arrays, all that are possible to be written - not just the ones that are actually filled with data by the user. **It writes empty arrays to**. 
It is inconsistant behavior camparing to other field types. For exemple, If a text field is localized, it writes itself to the database as object of localized objects, but only those which are actually filled with localized data. It does not writes empty objects. 
This behavior renders the fallback-locale mechanism failing to fill in the localised data for blocks field, as the mechanism is being notified that in particular object, there is an array based on locale, therefore it does not need fallback.

my blocks field definition:
```
{
  type:'blocks',
  name: 'blocks',
  label: 'Bloki z treścią',
  localized: true,
  blocks: [
    Richtext,
    Columns,
    GalleryColumns,
    LargePhoto,
  ]
},
```
none of the blocks have localized fields in them. Only the wrapper (blocks field) is localized.

## Other Details

```
"payload": "^1.2.0",
```

**Comment by jmikrut** [comment]
Hey @quornik — this is a good catch.

We discussed this as a team this morning and have identified what we need to do to resolve this.

Basically, an array / blocks field needs to be able to have either `null` as a value, which means that it has not been interacted with at all, _**or**_ an empty array to denote that it has specifically been assigned as empty. 

#### Case 1

You have created a document in a default locale, and then switched to a non-default locale. You have not tampered with a given array field at all, and assume that its value is blank, therefore it should be set as `null` in the database. This should then allow `fallbackLocale` to work.

#### Case 2

You have created a document in a default locale, and then switch to a non-default locale. You _**explicitly want**_ the given array field to retain its empty state in that non-default locale, and enforce that when a doc is retrieved in the non-default locale, that the array is empty. This would then need to store an empty array in the database.

Basically, we need both cases. I am thinking that it's best to store `null` by default, but the user would have to have a way to declare if the array field has a value, and it's empty, or it does not have a value at all. So we need some UI updates for this one.

We are going to address this shortly, via adding in controls for `array` and `blocks` field types, allowing you as the user to specify which state (`null` or `[]`) an array / blocks field should reside in.

**Comment by cherrmax** [comment]
> Hey @quornik — this is a good catch.
> 
> We discussed this as a team this morning and have identified what we need to do to resolve this.
> 
> Basically, an array / blocks field needs to be able to have either `null` as a value, which means that it has not been interacted with at all, _**or**_ an empty array to denote that it has specifically been assigned as empty. And this is only applicable if the field is localized. If not, then it doesn't apply (at least I don't think.)
> 
> #### Case 1
> You have created a document in a default locale, and then switched to a non-default locale. You have not tampered with a given array field at all, and assume that its value is blank, therefore it should be set as `null` in the database. This should then allow `fallbackLocale` to work.
> 
> #### Case 2
> You have created a document in a default locale, and then switch to a non-default locale. You _**explicitly want**_ the given array field to retain its empty state in that non-default locale, and enforce that when a doc is retrieved in the non-default locale, that the array is empty. This would then need to store an empty array in the database.
> 
> Basically, we need both cases. I am thinking that it's best to store `null` by default, but the user would have to have a way to declare if the array field has a value, and it's empty, or it does not have a value at all. So we need some UI updates for this one.
> 
> We are going to address this shortly, via adding in controls for `array` and `blocks` field types, allowing you as the user to specify which state (`null` or `[]`) an array / blocks field should reside in.

Thanks for looking at that
Just adding a bit from myself.
Got similar problem.
I have a block in the document with localized array. There are 2 locales.
I filled all the fields with both locales data. (2 text fields)
Switched to secondary locale, then re-arranged order of collapsible blocks in the array UI by dragging them up and down.
Then saved. Everything looked fine.
Then switched to default locale to find the array in that locale completely empty.

**Comment by cherrmax** [comment]
@jmikrut 
Thanks for looking at that
Just adding a bit from myself.
Got similar problem.
I have a block in the document with localized array. There are 2 locales.
I filled all the fields with both locales data. (2 text fields)
Switched to secondary locale, then re-arranged order of collapsible blocks in the array UI by dragging them up and down.
Then saved. Everything looked fine.
Then switched to default locale to find the array in that locale completely empty.
